model/riscv_sys_control.sail

```diff
@@ -61,8 +70,12 @@ function check_seed_CSR (csr : csreg, p : Privilege, isWrite : bool) -> bool = {
 function check_CSR(csr : csreg, p : Privilege, isWrite : bool) -> bool =
```

Comment:
Not for this PR, but wouldn't this be cleaner as a scattered function?

Reply:
I think this one shouldn't be a scattered function because `check_CSR_access()` is always the same, but the other functions should be moved into `is_CSR_defined()`, something like this:

```
function check_CSR(csr : csreg, p : Privilege, isWrite : bool) -> bool =
   check_CSR_access(csrAccess(csr), csrPriv(csr), p, isWrite) & is_CSR_defined(csr, p, isWrite)
```

Or we could potentially leave `isWrite` just for the `seed` CSR. It's the only one that needs it and unfortunately it is actually specified that way:

> Attempts to access the seed CSR using a read-only CSR-access instruction (CSRRS/CSRRC with rs1=x0 or CSRRSI/CSRRCI with uimm=0) raise an illegal instruction exception

I'll make a PR to clean this up after this one.